webaccel周りのリソースで意図しない差分が出る

### 不具合の説明と期待する動作

Acc Testを実施すると以下の差分が出る。
webaccelで同様の箇所が複数ある模様。

```
resource_sakuracloud_webaccel_test.go:372: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # sakuracloud_webaccel.foobar will be updated in-place
          ~ resource "sakuracloud_webaccel" "foobar" {
              - domain             = "xxxxx.user.webaccel.jp" -> null
                id                 = "xxxxx"
                name               = "dummy"
                # (8 unchanged attributes hidden)

              - cors_rules {
                  - allowed_origins = [
                      - "https://www2.example.com",
                      - "https://app.example.com",
                    ] -> null
                }
              + cors_rules {
                  + allow_all       = false
                  + allowed_origins = [
                      + "https://www2.example.com",
                      + "https://app.example.com",
                    ]
                }

              - origin_parameters {
                  # At least one attribute in this block is (or was) sensitive,
                  # so its contents will not be displayed.
                }
              + origin_parameters {
                  # At least one attribute in this block is (or was) sensitive,
                  # so its contents will not be displayed.
                }
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccSakuraCloudResourceWebAccel_InvalidConfigurations (44.21s)
```

### 再現手順

Acc Testの実施で再現

### 動作環境

```markdown
- ライブラリのバージョン: 最新版
- ツール・言語のバージョン: 最新版
- OSのバージョン: おそらくどのOSでも同じ
```

### 設定やコード

```markdown
なし
```

### ログ

```shell

```

--- a/sakuracloud/resource_sakuracloud_webaccel.go
+++ b/sakuracloud/resource_sakuracloud_webaccel.go
@@ -462,7 +462,9 @@ func setWebAccelResourceData(d *schema.ResourceData, client *APIClient, data *we
 func setWebAccelSiteResourceData(d *schema.ResourceData, client *APIClient, data *webaccel.Site) diag.Diagnostics {
 	d.Set("name", data.Name)                                              //nolint:errcheck,gosec
 	d.Set("domain_type", data.DomainType)                                 //nolint:errcheck,gosec
-	d.Set("domain", data.Domain)                                          //nolint:errcheck,gosec
+	if _, ok := d.GetOk("domain"); ok {
+		d.Set("domain", data.Domain) //nolint:errcheck,gosec
+	}
 	d.Set("subdomain", data.Subdomain)                                    //nolint:errcheck,gosec
 	d.Set("cname_record_value", data.Subdomain+".")                       //nolint:errcheck,gosec
 	d.Set("txt_record_value", fmt.Sprintf("webaccel=%s", data.Subdomain)) //nolint:errcheck,gosec

--- a/sakuracloud/resource_sakuracloud_webaccel.go
+++ b/sakuracloud/resource_sakuracloud_webaccel.go
@@ -462,7 +462,7 @@ func setWebAccelResourceData(d *schema.ResourceData, client *APIClient, data *we
 func setWebAccelSiteResourceData(d *schema.ResourceData, client *APIClient, data *webaccel.Site) diag.Diagnostics {
 	d.Set("name", data.Name)                                              //nolint:errcheck,gosec
 	d.Set("domain_type", data.DomainType)                                 //nolint:errcheck,gosec
-	if _, ok := d.GetOk("domain"); ok {
+	if data.DomainType == "own_domain" {
 		d.Set("domain", data.Domain) //nolint:errcheck,gosec
 	}
 	d.Set("subdomain", data.Subdomain)                                    //nolint:errcheck,gosec

--- a/sakuracloud/resource_sakuracloud_webaccel.go
+++ b/sakuracloud/resource_sakuracloud_webaccel.go
@@ -460,10 +460,12 @@ func setWebAccelResourceData(d *schema.ResourceData, client *APIClient, data *we
 	return setWebAccelSiteResourceData(d, client, data)
 }
 func setWebAccelSiteResourceData(d *schema.ResourceData, client *APIClient, data *webaccel.Site) diag.Diagnostics {
-	d.Set("name", data.Name)                                              //nolint:errcheck,gosec
-	d.Set("domain_type", data.DomainType)                                 //nolint:errcheck,gosec
+	d.Set("name", data.Name)              //nolint:errcheck,gosec
+	d.Set("domain_type", data.DomainType) //nolint:errcheck,gosec
 	if data.DomainType == "own_domain" {
 		d.Set("domain", data.Domain) //nolint:errcheck,gosec
+	} else {
+		d.Set("domain", "") //nolint:errcheck,gosec
 	}
 	d.Set("subdomain", data.Subdomain)                                    //nolint:errcheck,gosec
 	d.Set("cname_record_value", data.Subdomain+".")                       //nolint:errcheck,gosec

--- a/sakuracloud/resource_sakuracloud_webaccel.go
+++ b/sakuracloud/resource_sakuracloud_webaccel.go
@@ -465,7 +465,7 @@ func setWebAccelSiteResourceData(d *schema.ResourceData, client *APIClient, data
 	if data.DomainType == "own_domain" {
 		d.Set("domain", data.Domain) //nolint:errcheck,gosec
 	} else {
-		d.Set("domain", "") //nolint:errcheck,gosec
+		d.Set("domain", nil) //nolint:errcheck,gosec
 	}
 	d.Set("subdomain", data.Subdomain)                                    //nolint:errcheck,gosec
 	d.Set("cname_record_value", data.Subdomain+".")                       //nolint:errcheck,gosec

--- a/sakuracloud/structure_webaccel.go
+++ b/sakuracloud/structure_webaccel.go
@@ -91,13 +91,9 @@ func flattenWebAccelCorsRules(data []*webaccel.CORSRule) ([]interface{}, error) 
 			return nil, nil
 		}
 		corsRuleParams := make(map[string]interface{})
-		switch {
-		case rule.AllowsAnyOrigin:
-			corsRuleParams["allow_all"] = rule.AllowsAnyOrigin
-		case len(rule.AllowedOrigins) > 0:
+		corsRuleParams["allow_all"] = rule.AllowsAnyOrigin
+		if len(rule.AllowedOrigins) > 0 {
 			corsRuleParams["allowed_origins"] = rule.AllowedOrigins
-		default:
-			corsRuleParams["allow_all"] = false
 		}
 		return []interface{}{corsRuleParams}, nil
 	default:

--- a/sakuracloud/structure_webaccel_test.go
+++ b/sakuracloud/structure_webaccel_test.go
@@ -346,6 +346,7 @@ func TestFlattenWebAccelCorsRules(t *testing.T) {
 			},
 			[]interface{}{
 				map[string]interface{}{
+					"allow_all":       false,
 					"allowed_origins": []string{"origin1", "origin2"},
 				},
 			},

